### PR TITLE
config: add kubernetes jobs timeout env variables

### PR DESCRIPTION
**Pull request by VMois**
closes reanahub/reana-job-controller#343

### CODE REVIEW

**Comment by mvidalgarcia** [comment]
Needs rebase.

File: reana_workflow_controller/config.py [hunk 1]
```diff
@@ -95,6 +95,22 @@
...
+
+REANA_KUBERNETES_JOBS_MAX_USER_TIMEOUT_LIMIT = os.getenv(
+    "REANA_KUBERNETES_JOBS_MAX_USER_TIMEOUT_LIMIT"
+)
+"""Maximum custom timeout in seconds that users can assign to their job.
+
+Please see the following URL for more details
+https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup.
+"""
+
```
Comment (mvidalgarcia) [inline reana_workflow_controller/config.py:113]: The same [comment](https://github.com/reanahub/reana/pull/610#discussion_r788621722) from the `reana` PR applies here.

**Review by mvidalgarcia — APPROVED** [review]
LGTM!

Just a minor comment, other than that works as expected.

**Comment by VMois** [comment]
Should I update CHANGELOG here?

**Comment by mvidalgarcia** [comment]
> Should I update CHANGELOG here?

The change is not that relevant in this PR, I wouldn't update it. The main changelog in the reana repo kind of covers it. But I have no strong opinion.

PS. Needs rebase.

**Comment by tiborsimko** [comment]
FWIW I would advocate for always updating the CHANGES file, even if changes are sort of "internal" and the main user facing functionality is covered elsewhere; simply because the component will be released separately, and we'll have to put _something_ for 0.8.1 release notes soon  there anyway :smile: So having the raw notes already in the file with each PR will greatly simplify the release note compilation work later.

**Review by tiborsimko — APPROVED** [review]

**Pull request merged**